adcc/misc.py

```diff
@@ -47,6 +48,38 @@ def get(self):
     return property(get)
 
 
+def cached_member_function(function):
+    """
+    Decorates a member function being called with
+    one or more arguments and stores the results
+    in field `_function_cache` of the class instance.
+    """
+    fname = function.__name__
+
+    @wraps(function)
+    def wrapper(*args):
+        self = args[0]
+        key = args[1:]
+        if not hasattr(self, "_function_cache"):
+            self._function_cache = {}
+        cache = self._function_cache
+        if fname not in cache:
+            cache[fname] = {}
```

Comment:
The pythonic way is to catch the exceptions here, see the implementation of the function above. Typically gives rise to less code.

Reply:
I think in this case this may lead to too many `try` & `except` in a row, because we have to take care of a) the function and b) the correct parameters. I can give it a try, but I'm not a huge fan of the exception stuff.

Reply:
Done with `try/except` now 👍 I like it better this way.